dice_ml/explainer_interfaces/dice_genetic.py

```diff
@@ -272,6 +264,15 @@ def _generate_counterfactuals(self, query_instance, total_CFs, initialization="k
                 query_instance=query_instance_orig)
         query_instance = self.data_interface.prepare_query_instance(
                 query_instance=query_instance)
+        # number of output nodes of ML model
+        self.num_output_nodes = None
+        if self.model.model_type == ModelTypes.Classifier:
+            self.num_output_nodes = self.model.get_num_output_nodes2(query_instance)
+
+        # number of output nodes of ML model
```

Comment:
Duplicate code here?

Reply:
ah yes, good catch. this needs to be removed.